Zend/zend_closures.c

```diff
@@ -424,10 +423,22 @@ static HashTable *zend_closure_get_debug_info(zval *object, int *is_temp) /* {{{
 	ALLOC_HASHTABLE(debug_info);
 	zend_hash_init(debug_info, 8, NULL, ZVAL_PTR_DTOR, 0);
 
-	if (closure->func.type == ZEND_USER_FUNCTION && closure->func.op_array.static_variables) {
-		HashTable *static_variables = closure->func.op_array.static_variables;
-		ZVAL_ARR(&val, zend_array_dup(static_variables));
+	if (closure->func.type == ZEND_USER_FUNCTION && closure->func.op_array.last_static_var) {
+		zval *entry;
+		uint32_t i = 0;
+
+		array_init(&val);
 		zend_hash_str_update(debug_info, "static", sizeof("static")-1, &val);
+		for (; i < closure->func.op_array.last_static_var; i++) {
+			entry = &closure->func.op_array.static_variables[i];
+			if (Z_TYPE_P(entry) == IS_REFERENCE &&
+				Z_REFCOUNT_P(entry) == 1 && 
+				Z_TYPE_P(Z_REFVAL_P(entry)) != IS_ARRAY) {
```

Comment:
I still don't really understand it. The code in zend_hash covers the case where the array references the containing array, which here would have been the static variable table. However such a reference is not possible, so the condition should not be relevant for us. The call_user_func_003.phpt test doesn't deal with arrays at all.